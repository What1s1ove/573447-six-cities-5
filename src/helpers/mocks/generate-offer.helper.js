import {getRandomItem} from '~/helpers/array/array';
import {getRandomNumber} from '~/helpers/number/number';
import {generateRandomId} from '~/helpers/mocks/generate-random-id.helper';
import {OfferCity, OfferType} from '~/common/enums/enums';

const TITLES = [
  `Beautiful \u0026 luxurious apartment at great location`,
  `Wood and stone place`,
  `Canal View Prinsengracht`,
  `Nice, cozy, warm big bed apartment`,
];

const OfferConfig = {
  PRICE: {
    MIN: 50,
    MAX: 250,
  },
  RATING: {
    MIN: 0,
    MAX: 5,
    FRACTIONAL_PART_COUNT: 1,
  },
};

const cities = Object.values(OfferCity);
const offerTypes = Object.values(OfferType);

const generateOffer = () => ({
  id: generateRandomId(),
  city: getRandomItem(cities),
  type: getRandomItem(offerTypes),
  title: getRandomItem(TITLES),
  price: getRandomNumber(OfferConfig.PRICE.MIN, OfferConfig.PRICE.MAX),
  rating: getRandomNumber(
      OfferConfig.RATING.MIN,
      OfferConfig.RATING.MAX,
      OfferConfig.RATING.FRACTIONAL_PART_COUNT
  ),
});

export {generateOffer};
