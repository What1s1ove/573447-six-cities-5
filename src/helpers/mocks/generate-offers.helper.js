import {generateOffer} from '~/helpers/mocks/generate-offer.helper';

const generateOffers = (count) => {
  const offers = Array.from(new Array(count), generateOffer);

  return offers;
};

export {generateOffers};
