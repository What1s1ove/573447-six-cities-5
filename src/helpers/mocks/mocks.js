export * from './generate-offer.helper';
export * from './generate-offers.helper';
export * from './generate-random-id.helper';
