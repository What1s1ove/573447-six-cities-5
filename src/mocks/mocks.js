export * from './offers/offers';
export * from './user/user';
export * from './reviews/reviews';
export * from './city/city';
export * from './store/store';
