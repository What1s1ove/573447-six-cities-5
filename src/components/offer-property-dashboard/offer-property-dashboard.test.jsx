import * as React from 'react';
import {render} from 'enzyme';
import {mockedOffer} from '~/mocks/mocks';
import OfferPropertyDashboard from './offer-property-dashboard';

describe(`OfferPropertyDashboard component`, () => {
  it(`should render OfferPropertyDashboard component`, () => {
    const component = render(
        <OfferPropertyDashboard
          offer={mockedOffer}
          onFavoriteToggle={jest.fn()}
        />
    );

    expect(component).toMatchSnapshot();
  });
});
