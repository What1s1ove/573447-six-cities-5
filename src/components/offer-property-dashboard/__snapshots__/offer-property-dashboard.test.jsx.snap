// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferPropertyDashboard component should render OfferPropertyDashboard component 1`] = `
Array [
  <div
    class="property__mark"
  >
    <span>
      Premium
    </span>
  </div>,
  <div
    class="property__name-wrapper"
  >
    <h1
      class="property__name"
    >
      The Joshua Tree House
    </h1>
    <button
      class="property__bookmark-button button"
      type="button"
    >
      <svg
        class="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          href="#icon-bookmark"
        />
      </svg>
      <span
        class="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    class="property__rating rating"
  >
    <div
      class="property__stars rating__stars"
    >
      <span
        style="width:80%"
      />
      <span
        class="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      class="property__rating-value rating__value"
    >
      4.4
    </span>
  </div>,
  <ul
    class="property__features"
  >
    <li
      class="property__feature property__feature--entire"
    >
      Private Room
    </li>
    <li
      class="property__feature property__feature--bedrooms"
    >
      1 Bedrooms
    </li>
    <li
      class="property__feature property__feature--adults"
    >
      Max 2 adults
    </li>
  </ul>,
  <div
    class="property__price"
  >
    <b
      class="property__price-value"
    >
      €241
    </b>
    <span
      class="property__price-text"
    >
       night
    </span>
  </div>,
  <div
    class="property__inside"
  >
    <h2
      class="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      class="property__inside-list"
    >
      <li
        class="property__inside-item"
      >
        Dishwasher
      </li>
      <li
        class="property__inside-item"
      >
        Towels
      </li>
      <li
        class="property__inside-item"
      >
        Washer
      </li>
      <li
        class="property__inside-item"
      >
        Fridge
      </li>
      <li
        class="property__inside-item"
      >
        Breakfast
      </li>
      <li
        class="property__inside-item"
      >
        Laptop friendly workspace
      </li>
      <li
        class="property__inside-item"
      >
        Baby seat
      </li>
      <li
        class="property__inside-item"
      >
        Air conditioning
      </li>
    </ul>
  </div>,
  <div
    class="property__host"
  >
    <h2
      class="property__host-title"
    >
      Meet the host
    </h2>
    <div
      class="property__host-user user"
    >
      <div
        class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
      >
        <img
          alt="Host avatar"
          class="property__avatar user__avatar"
          height="74"
          src="img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        class="property__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      class="property__description"
    >
      <p
        class="property__text"
      >
        This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.
      </p>
    </div>
  </div>,
]
`;
